src/pages/docs/creators/ray/mainnet.md

```diff
@@ -0,0 +1,105 @@
+
+When your application is getting more complex, and you need more power, you should move to the mainnet - there are more nodes, and the nodes themselves are more powerful, but you have to pay to use them.
+
+There are a couple of considerations when entering the mainnet:
+- [Mainnet in the cluster yaml](#mainnet-in-the-cluster-yaml) - switching to the mainnet is as simple as changing on property in the cluster yaml (besides getting funds)
+- [Payments](#mainnet-payments) - operating on the mainnet might be cheap, but you still need to pay the providers running your Ray on Golem cluster
+- [Funds onboarding](#mainnet-funds-onboarding) - when you don't have mainnet funds, Ray on Golem will guide you to get them
+- [Budget Control](#budget-control) - Ray on Golem provides a couple of mechanisms to let you control how much you spend
+- [Finding providers](#finding-providers) - the Golem Network mainnet is peer-to-peer, meaning that the more nodes there are the more time it is required for information about them to reach you
+
+
+Check out [the online providers](https://stats.golem.network/network/providers/online) if you are curious about what kind of machines are available on the mainnet.
+
+
+## Mainnet in the cluster yaml
+
+Switching to the mainnet is as easy as changing the `provider.parameters.network` property in your [cluster yaml](/docs/creators/ray/cluster-yaml):
+- `network: "goerli"` stands for the testnet
+- `network: "polygon"` stands for the mainnet
+
+If you have funds this is the only change you are required to make to start your Ray on Golem cluster on the mainnet - the `ray up` will know what to do.
+
+If you don't have any funds, `ray up` will fail and direct you to the onboarding portal which will guide you to getting them.
+
+Please consider using [budget control](#budget-control) mechanisms to manage your spending.
+
+## Mainnet payments
+
+Payments within the Golem Network take place on the [Polygon](https://polygon.technology) blockchain and require some GLM tokens (to pay the providers) and MATIC (for payment transaction fees).
+
+Feel free to learn more about general [mainnet payments and funding your Golem node](/docs/creators/javascript/guides/switching-to-mainnet).
+
+Note that the funds you spend on your Ray on Golem cluster don't go to Golem Factory. The GLMs go directly to providers you rent the nodes from and the MATICs go directly to Polygon blockchain validators who enable your payments.
+
+## Mainnet funds onboarding
+
+When your cluster yaml sports `network: "polygon"` your `ray up` attempts to run the cluster on the Golem mainnet.
+
+If you don't have the mainnet funds, it will fail and print a link to our onboarding portal.
+
+```bash
+ray up golem-cluster.yaml --yes
+```
+```
+(todo: ray up output with link to onboarding portal)
+```
+
+The onboarding link is also printed in case you have the mainnet funds (it won't cause `ray up` to fail in this case) - so you can always top up your wallet.
```

Comment:
```suggestion
Even if you do have the required funds, `ray up` will nevertheless print the onboarding link as a reminder, in case you'd like to top up your wallet.
```